src/app/pages/SpotTradingPage/components/TradeForm/LimitForm.tsx

```diff
@@ -0,0 +1,311 @@
+import { useAssetBalanceOf } from '../../../../hooks/useAssetBalanceOf';
+import { bignumber } from 'mathjs';
+import { useWalletContext } from '@sovryn/react-wallet';
+import { TradingTypes, ITradeFormProps, IApiLimitOrder } from '../../types';
+import { OrderType } from 'app/components/OrderTypeTitle/types';
+import { AssetRenderer } from 'app/components/AssetRenderer';
+import { maxMinusFee } from 'utils/helpers';
+import { stringToFixedPrecision } from 'utils/display-text/format';
+import { AvailableBalance } from 'app/components/AvailableBalance';
+import { ErrorBadge } from 'app/components/Form/ErrorBadge';
+import { useMaintenance } from 'app/hooks/useMaintenance';
+import { discordInvite, tradeFormsGasLimit } from 'utils/classifiers';
+import styles from './index.module.scss';
+import { Duration } from '../LimitOrderSetting/Duration';
+import { TradeDialog } from '../TradeDialog';
+import { OrderLabel } from '../OrderLabel';
+import { useLimitOrder } from 'app/hooks/limitOrder/useLimitOrder';
+import classNames from 'classnames';
+import { useSwapsExternal_getSwapExpectedReturn } from 'app/hooks/swap-network/useSwapsExternal_getSwapExpectedReturn';
+import { toWei } from 'web3-utils';
+import { weiToFixed } from 'utils/blockchain/math-helpers';
+import { actions } from '../../slice';
+import { useDispatch } from 'react-redux';
+import { Toast } from 'app/components/Toast';
+import { TransactionDialog } from 'app/components/TransactionDialog';
+import { TxStatus } from 'store/global/transactions-store/types';
+import { LimitResultDialog } from './LimitResultDialog';
+
+export const LimitForm: React.FC<ITradeFormProps> = ({
+  sourceToken,
+  targetToken,
+  tradeType,
+  hidden,
+}) => {
+  const { t } = useTranslation();
+  const { connected } = useWalletContext();
+  const { checkMaintenance, States } = useMaintenance();
+  const spotLocked = checkMaintenance(States.SPOT_LIMIT);
+  const dispatch = useDispatch();
+
+  const [tradeDialog, setTradeDialog] = useState(false);
+
+  const [orderStatus, setOrderStatus] = useState(TxStatus.NONE);
+  const [txHash, setTxHash] = useState('');
+
+  const [amount, setAmount] = useState('');
+  const [limitPrice, setLimitPrice] = useState('');
+
+  const [duration, setDuration] = useState(0);
+
+  const amountOut = useMemo(() => {
+    if (!limitPrice || !amount || limitPrice === '0') return '0';
+
+    return bignumber(amount || '0')
+      .times(limitPrice)
+      .toString();
+  }, [limitPrice, amount]);
+
+  const weiAmount = useWeiAmount(amount);
+  const weiAmountOut = useWeiAmount(amountOut);
+
+  const { value: balance } = useAssetBalanceOf(sourceToken);
+  const { value: marketPrice } = useSwapsExternal_getSwapExpectedReturn(
+    sourceToken,
+    targetToken,
+    toWei('1'),
+  );
+
+  const limitMarketChange = useMemo(() => {
+    if (!limitPrice || limitPrice === '0' || !marketPrice) return '';
+
+    return bignumber(limitPrice)
+      .div(weiToFixed(marketPrice, 6))
```

Comment:
It's weird that you apply truncation in calculation. I think if you want to have it formatted somehow, it should happen at the place of usage, not here.

Reply:
fixed
@Rickk137 @rick23p  check pls to make sure that everything is working correctly